### PR TITLE
Added support for Laravel 10

### PR DESCRIPTION
**Pull request by Vincent-dejong**

### CODE REVIEW

**Comment by rolfdenhartog** [comment]
Nice! I think `.github/workflows/main.yml` should also be updated containing PHP 8.2. I'm not sure how I can add that to this PR.

https://github.com/fedeisas/laravel-mail-css-inliner/blob/6f98d7425e2c95032beae2885f3ca104079ab482/.github/workflows/main.yml

**Comment by fabiofdsantos** [comment]
ping @DannyvdSluijs

File: composer.json [hunk 1]
```diff
@@ -16,8 +16,8 @@
     "require": {
         "php": "^8.0.2",
         "ext-dom": "*",
-        "illuminate/mail": "^9.0",
-        "illuminate/support": "^9.0",
+        "illuminate/mail": "^9.0||^10.0",
+        "illuminate/support": "^9.0||^10.0.0",
```
Comment (DannyvdSluijs) [inline composer.json:20]: Could you make this inline with the other constraints? Like ^10.0

**Comment by DannyvdSluijs** [comment]
In addition there is some feedback in the pipeline see https://github.com/fedeisas/laravel-mail-css-inliner/actions/runs/4182613296/jobs/7356352244

File: composer.json (hunk 1 above)
Comment (ABartelt) [reply composer.json:20]:
```suggestion
      "illuminate/support": "^9.0||^10.0",
```

**Comment by TSchuermans** [comment]
@DannyvdSluijs doesn't that mean there has to be a space surrounding the || in the version constraint?

**Comment by driftingly** [comment]
> @DannyvdSluijs doesn't that mean there has to be a space surrounding the || in the version constraint?

@TSchuermans @Vincent-dejong Looks that way to me

**Comment by driftingly** [comment]
Laravel 10 also requires PHP 8.1.0 or greater.
https://laravel.com/docs/10.x/upgrade#updating-dependencies

File: composer.json [hunk 2]
```diff
@@ -16,8 +16,8 @@
     "require": {
         "php": "^8.0.2",
         "ext-dom": "*",
-        "illuminate/mail": "^9.0",
-        "illuminate/support": "^9.0",
+        "illuminate/mail": "^9.0||^10.0",
+        "illuminate/support": "^9.0||^10.0",
```
Comment (DannyvdSluijs) [inline composer.json:20]:
```suggestion
        "illuminate/support": "^9.0 || ^10.0",
```

File: composer.json [hunk 3]
```diff
@@ -16,8 +16,8 @@
     "require": {
         "php": "^8.0.2",
         "ext-dom": "*",
-        "illuminate/mail": "^9.0",
-        "illuminate/support": "^9.0",
+        "illuminate/mail": "^9.0||^10.0",
```
Comment (DannyvdSluijs) [inline composer.json:19]:
```suggestion
        "illuminate/mail": "^9.0 || ^10.0",
```

**Comment by Vincent-dejong** [comment]
Thanks everyone! I'm on holiday so I missed the feedback but cool to see it's now merged.

**Comment by kirkbushell** [comment]
It's not actually working on Laravel 10. Was this tested?

**Comment by DannyvdSluijs** [comment]
@kirkbushell 

I can't recall completely, but I would assume so. If you can share a repo, gist or just bits of code to reproduce we can take it from there. Feel free to open a new issue for this. As a closed Pull Request might not get the attention this deserves.

**Pull request merged**